napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -149,3 +269,91 @@ def test_constructor_visible_widgets(plugin_dialog_constructor):
     """
     assert not plugin_dialog_constructor.direct_entry_edit.isVisible()
     assert not plugin_dialog_constructor.direct_entry_btn.isVisible()
+
+
+def test_version_dropdown(plugin_dialog):
+    """
+    Test that when the source drop down is changed, it displays the other versions properly.
+    """
+    widget = plugin_dialog.available_list.item(1).widget
+    assert widget.version_choice_dropdown.currentText() == "3"
+    # switch from PyPI source to conda one.
+    widget.source_choice_dropdown.setCurrentIndex(1)
+    assert widget.version_choice_dropdown.currentText() == "4.5"
+
+
+def test_plugin_list_count_items(plugin_dialog):
+    assert plugin_dialog.installed_list._count_visible() == 2
+
+
+def test_plugin_list_handle_action(plugin_dialog, qtbot):
+    item = plugin_dialog.installed_list.item(0)
+
+    plugin_dialog.installed_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.UPGRADE,
+    )
+
+    with patch.object(qt_plugin_dialog.WarnPopup, "exec_") as mock:
+        plugin_dialog.installed_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.UNINSTALL,
+        )
+        assert mock.called
+
+    item = plugin_dialog.available_list.item(0)
+    plugin_dialog.available_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.INSTALL,
+        version='3',
+    )
+
+    plugin_dialog.available_list.handle_action(
+        item, 'test-name-1', InstallerActions.CANCEL, version='3'
+    )
```

Comment:
What is tested here? I do not see checking if cancel happens correctly.

Reply:
I updated all the tests here to make sure that the set_busy function is called with the proper action.  what do you think?